src/ConnectFour/Port/Adapter/Persistence/Repository/SelectShardGameRepository.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Gaming\ConnectFour\Port\Adapter\Persistence\Repository;
+
+use Doctrine\DBAL\Connection;
+use Gaming\ConnectFour\Domain\Game\Game;
+use Gaming\ConnectFour\Domain\Game\GameId;
+use Gaming\ConnectFour\Domain\Game\Games;
+use mysqli;
+
+final class SelectShardGameRepository implements Games
+{
+    public function __construct(
+        private readonly Games $games,
+        private readonly Connection $connection,
+        private readonly string $databaseNamePrefix,
+        private readonly int $numberOfShards
+    ) {
+    }
+
+    public function nextIdentity(): GameId
+    {
+        return $this->games->nextIdentity();
+    }
+
+    /**
+     * @throw ConcurrencyException
+     */
+    public function save(Game $game): void
+    {
+        $this->selectDatabase($game->id());
+
+        $this->games->save($game);
+    }
+
+    public function get(GameId $id): Game
+    {
+        $this->selectDatabase($id);
+
+        return $this->games->get($id);
+    }
+
+    private function selectDatabase(GameId $gameId): void
+    {
+        $mysqli = $this->connection->getNativeConnection();
+        assert($mysqli instanceof mysqli);
+
+        $mysqli->select_db(
+            $this->databaseNamePrefix . ((crc32($gameId->toString()) % $this->numberOfShards) + 1)
```

Comment:
This algorithm, although not deem necessary for this project, could be replaced by something more smart, e.g. [consistent hashing](https://en.wikipedia.org/wiki/Consistent_hashing). Consistent hashing would reduce the likelihood of games being reassigned to another shard in the event of resharding.

Reply:
Structure changed, new comment made.